includes/class-newspack-popups.php

```diff
@@ -941,21 +941,62 @@ public static function archive_campaign( $id, $status ) {
 		update_term_meta( $id, self::NEWSPACK_POPUPS_TAXONOMY_STATUS, $status ? 'archive' : '' );
 	}
 
+	/**
+	 * Remove the default category from the given post, if it's the only category applied to that post.
+	 *
+	 * @param int $post_id ID of the post.
+	 */
+	private static function remove_default_category( $post_id ) {
+		$default_category_id = (int) get_option( 'default_category', false );
+		$post_categories     = wp_get_post_categories( $post_id );
+		if ( 1 === count( $post_categories ) && reset( $post_categories ) === $default_category_id ) {
+			wp_remove_object_terms( $post_id, $default_category_id, 'category' );
+		}
+	}
+
 	/**
 	 * Prevent setting the default category when publishing.
 	 *
 	 * @param string $new_status New status.
 	 * @param string $old_status Old status.
 	 * @param bool   $post Post.
 	 */
-	public static function remove_default_category( $new_status, $old_status, $post ) {
+	public static function prevent_default_category_on_publish( $new_status, $old_status, $post ) {
 		if ( self::NEWSPACK_POPUPS_CPT === $post->post_type && 'publish' !== $old_status && 'publish' === $new_status ) {
-			$default_category_id = (int) get_option( 'default_category', false );
-			$popup_has_category  = has_category( $default_category_id, $post->ID );
-			if ( $popup_has_category ) {
-				wp_remove_object_terms( $post->ID, $default_category_id, 'category' );
-			}
+			self::remove_default_category( $post->ID );
+		}
+	}
+
+	/**
+	 * When a category is deleted, any posts that have only that category assigned
+	 * are automatically assigned the site's default category (usually "Uncategorized").
+	 * We want to prevent this behavior for prompts, as prompts with the default
+	 * category will only appear on posts with that category.
+	 *
+	 * @return int The number of prompts affected by this callback.
+	 */
+	public static function prevent_default_category_on_term_delete() {
+		$prompts_updated          = 0;
+		$default_category_id      = (int) get_option( 'default_category', false );
+		$default_category_prompts = get_posts(
+			[
+				'post_status'    => 'any',
+				'post_type'      => self::NEWSPACK_POPUPS_CPT,
+				'posts_per_page' => -1,
+				'tax_query'      => [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+					'taxonomy'         => 'category',
+					'terms'            => $default_category_id,
```

Comment:
```suggestion
					'terms'            => [ $default_category_id, $deleted_category_id ],
```

I did not test this, but it's somewhat the idea I had so it doesn't remove the term relation of every prompt with the default category.

Reply:
Unfortunately this won't work because the [`delete_{$taxonomy}`](https://developer.wordpress.org/reference/hooks/delete_taxonomy/) hook fires _after_ the term is deleted, so running a `WP_Query` with its term ID will return every post.

Reply:
Looks like we can use [`pre_delete_term`](https://developer.wordpress.org/reference/hooks/pre_delete_term/) instead: I'll try that!